contracts/gov/src/staking.rs

```diff
@@ -1,33 +1,36 @@
 use crate::error::ContractError;
 use crate::state::{
-    bank_read, bank_store, config_read, config_store, poll_read, poll_voter_store, state_read,
-    state_store, Config, Poll, State, TokenManager,
+    bank_read, bank_store, config_read, config_store, is_synced_read, is_synced_store, poll_read,
+    poll_voter_store, state_read, state_store, Config, Poll, State,
+};
+use crate::voting_escrow::{
+    generate_extend_lock_amount_message, generate_extend_lock_time_message,
+    generate_withdraw_message,
 };
 
 use anchor_token::gov::{PollStatus, StakerResponse};
 use astroport::querier::query_token_balance;
 use cosmwasm_std::{
-    to_binary, Addr, CanonicalAddr, CosmosMsg, Deps, DepsMut, MessageInfo, Response, StdResult,
-    Storage, Uint128, WasmMsg,
+    to_binary, CanonicalAddr, CosmosMsg, Deps, DepsMut, MessageInfo, Response, StdResult, Uint128,
+    WasmMsg,
 };
 use cw20::Cw20ExecuteMsg;
 
-pub fn stake_voting_tokens(
+// can only be called when user's unlock_time > current_time
```

Comment:
this function will call `extend_lock_amount` of ve contract, which will do this check. i don't it's necessary to do this check here because we don't know the user's unlock time unless query ve contract, and if we do so, why not just do the check in `extend_lock_amount` of ve contract.

Reply:
sounds good!